src/unlock/rc_data.rs

```diff
@@ -0,0 +1,205 @@
+use lazy_static::lazy_static;
+
+use sparse_merkle_tree::{default_store::DefaultStore, SparseMerkleTree, H256 as SmtH256};
+
+use crate::types::xudt_rce_mol::{
+    RCDataBuilder, RCDataUnion, RCRuleBuilder, SmtProofBuilder, SmtProofEntryBuilder,
+    SmtProofEntryVec, SmtProofEntryVecBuilder,
+};
+use bytes::Bytes;
+use ckb_hash::{new_blake2b, Blake2b};
+use ckb_types::{molecule, prelude::*};
+use sparse_merkle_tree::traits::Hasher;
+
+lazy_static! {
+    pub static ref SMT_EXISTING: SmtH256 = SmtH256::from([
+        1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0,
+    ]);
+    pub static ref SMT_NOT_EXISTING: SmtH256 = SmtH256::from([
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0,
+    ]);
+}
+
+type SMT = SparseMerkleTree<CKBBlake2bHasher, SmtH256, DefaultStore<SmtH256>>;
+pub type Result<T> = ::core::result::Result<T, RcDataError>;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum RcDataError {
+    BuildTree(String),
+    CompileProof(String),
+}
```

Comment:
Use `thieerror::Error` to improve the error type, https://github.com/nervosnetwork/ckb-sdk-rust/blob/f141d9ff2af6895203ab3ab8663ab2f0be7bd3f2/src/traits/mod.rs#L35-L49